src/components/CommandBar/CommandBarSelectionMixedInput.tsx

```diff
@@ -63,14 +63,32 @@ export default function CommandBarSelectionMixedInput({
         setHasAutoSkipped(true)
       }
     }
-  }, [])
+  }, [arg.name])
 
   // Set selection filter if needed, and reset it when the component unmounts
   useEffect(() => {
     arg.selectionFilter && kclManager.setSelectionFilter(arg.selectionFilter)
     return () => kclManager.defaultSelectionFilter(selection)
   }, [arg.selectionFilter])
 
+  // Watch for outside teardowns of this component
+  // (such as clicking another argument in the command palette header)
+  // and quickly save the current selection if we can
+  useEffect(() => {
+    return () => {
+      const resolvedSelection: Selections | undefined = isArgRequired
+        ? selection
+        : selection || {
+            graphSelections: [],
+            otherSelections: [],
+          }
+
+      if (canSubmitSelection && resolvedSelection) {
+        onSubmit(resolvedSelection)
+      }
+    }
+  }, [])
```

Comment:
@franknoirot do we need the hasClearedSelection logic here as well?

Reply:
Yeah but [the `selectionMixed` argument inputType doesn't support the `clearSelectionFirst` flag](https://github.com/KittyCAD/modeling-app/blob/revert-7048-franknoirot/adhoc/just-use-modify-with-text-to-cad/src/lib/commandTypes.ts#L189-L204). It would if that did. I don't like that they've already diverged, so I think we should get rid of `selectionMixed` as soon as we can and just make the `selection` type support code-based selection in the future, but I don't want to do that in this PR. What do you think?

Reply:
Oh nice, yes, that makes sense!